File: fetch-open-prs.js
#!/usr/bin/env node

import https from 'https';
import fs from 'fs';

// GitHub API endpoint for open PRs
const repo = 'Zion-Holdings/zion.app';
const url = `https://api.github.com/repos/${repo}/pulls?state=open&per_page=100`;

console.log('Fetching open PRs from GitHub...');

const options = {
  headers: {
    'User-Agent': 'GitHub-PR-Merger',
    'Accept': 'application/vnd.github.v3+json'
  }
};

https.get(url, options, (res) => {
  let data = '';

  res.on('data', (chunk) => {
    data += chunk;
  });

  res.on('end', () => {
    try {
      const prs = JSON.parse(data);
      
      if (prs.length === 0) {
        console.log('No open PRs found.');
        return;
      }

      console.log(`Found ${prs.length} open PRs:`);
      console.log('=====================================');
      
      prs.forEach((pr, index) => {
        console.log(`${index + 1}. PR #${pr.number}: ${pr.title}`);
        console.log(`   Branch: ${pr.head.ref}`);
        console.log(`   Author: ${pr.user.login}`);
        console.log(`   Created: ${pr.created_at}`);
        console.log(`   URL: ${pr.html_url}`);
        console.log(`   Mergeable: ${pr.mergeable === null ? 'Unknown' : pr.mergeable}`);
        console.log('   ---');
      });

      // Save to file for processing
      fs.writeFileSync('open-prs.json', JSON.stringify(prs, null, 2));
      console.log('\nOpen PRs saved to open-prs.json');
      
    } catch (error) {
      console.error('Error parsing PR data:', error.message);
    }
  });
}).on('error', (error) => {
  console.error('Error fetching PRs:', error.message);
});

File: fetch_open_prs.py
#!/usr/bin/env python3
import requests
import json
import os

# GitHub API configuration
GITHUB_TOKEN = os.getenv('GITHUB_TOKEN', 'ghs_WA7UKPiDq0iKDQ5T2mdlyPqvDQDKv03Rjut5')
REPO_OWNER = 'Zion-Holdings'
REPO_NAME = 'zion.app'

def fetch_open_prs():
    """Fetch all open pull requests"""
    url = f"https://api.github.com/repos/{REPO_OWNER}/{REPO_NAME}/pulls"
    headers = {
        'Authorization': f'token {GITHUB_TOKEN}',
        'Accept': 'application/vnd.github.v3+json'
    }
    params = {'state': 'open', 'per_page': 100}
    
    try:
        response = requests.get(url, headers=headers, params=params)
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error fetching PRs: {e}")
        return []

def main():
    prs = fetch_open_prs()
    
    print(f"Found {len(prs)} open pull requests:")
    print("=" * 50)
    
    for pr in prs:
        print(f"PR #{pr['number']}: {pr['title']}")
        print(f"  Branch: {pr['head']['ref']}")
        print(f"  State: {pr['state']}")
        print(f"  Mergeable: {pr.get('mergeable', 'unknown')}")
        print(f"  Mergeable State: {pr.get('mergeable_state', 'unknown')}")
        print(f"  URL: {pr['html_url']}")
        print("-" * 30)
    
    # Save to file for further processing
    with open('/workspace/open_prs.json', 'w') as f:
        json.dump(prs, f, indent=2)
    
    print(f"\nSaved {len(prs)} PRs to /workspace/open_prs.json")

if __name__ == "__main__":
    main()